### PR TITLE
Backfill trn to teacher profile

### PR DESCRIPTION
**Pull request by slawosz**

### CODE REVIEW

File: app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        if trns.count == 1 && trns.first.length == 7
+          profile.trn = trns.first
+          profile.save!
+        end
+      end
+    end
+
+  private
+
+    def incorrect_teacher_profiles
```
Comment (peteryates) [inline app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb:19]: Are these incorrect or missing?

Comment (peteryates) [inline app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb:19]:
```suggestion
    def teacher_profiles_without_trns
```

File: app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Oneoffs::NPQ
+  class BackfillTrnToTeacherProfile
+    def migrate
+      # locate all TeacherProfiles with empty TRN
+      incorrect_teacher_profiles.in_batches.each_record do |profile|
+        trns = profile.user.npq_applications.map(&:teacher_reference_number).uniq
+        # check if they are correct
+        if trns.count == 1 && trns.first.length == 7
```
Comment (tonyheadford) [inline app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb:10]: So we have this to help ensure that only valid, correctly formatted TRNs are stored on the `TeacherProfile` - could that be used here perhaps given you're checking?

File: app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Oneoffs::NPQ
+  class BackfillTrnToTeacherProfile
+    def migrate
+      # locate all TeacherProfiles with empty TRN
+      incorrect_teacher_profiles.in_batches.each_record do |profile|
```
Comment (tonyheadford) [inline app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb:7]: Needs updating in line with the name change
```suggestion
      teacher_profiles_without_trns.in_batches.each_record do |profile|
```

File: app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  class BackfillTrnToTeacherProfile
+    def migrate
+      # locate all TeacherProfiles with empty TRN
+      teacher_profiles_without_trns.in_batches.each_record do |profile|
+        trns = profile.user.npq_applications.map(&:teacher_reference_number).uniq
+        # check if they are correct
+        if trns.count == 1 && TeacherReferenceNumber.new(trns.first).valid?
+          profile.trn = trns.first
+          profile.save!
+        end
```
Comment (tonyheadford) [inline app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb:13]: Can we use the `TeacherReferenceNumber.formatted_trn` here to ensure we store a correctly formatted one? 
```suggestion
        trn = TeacherReferenceNumber.new(trns.first)
        if trns.count == 1 && trn.valid?
          profile.trn = trn.formatted_trn
          profile.save!
        end
```

**Review by tonyheadford — APPROVED** [review]

**Pull request merged**